v_4_9_0/worker_template.go

```diff
@@ -138,15 +138,32 @@ systemd:
       ExecStopPost=-/usr/bin/docker rm -f $NAME
       [Install]
       WantedBy=multi-user.target
+  - name: k8s-setup-download-hyperkube.service
```

Comment:
Note: We want to change this later on, for now this is the simplest thing to do.

Reply:
Why is it bad? What are the ideas?

Reply:
This causes us to pull the entire hyperkube image just to then only use the binaries. It would be much better to only pull the binaries.

Reply:
It's mainly bad because Docker isn't intended as a software distribution platform. Alternatives include downloading binaries directly from Google (might have issues on nodes in China), baking it into our AMIs (not sure how this works on Azure/KVM), uploading/downloading via S3 (maybe proxied through giantswarm.io?).

Reply:
It would have been great to bake it into our AMI but we don't maintain AMIs. We simply pull CoreOS AMIs as they are and use them and then configure accordingly. I agree anyway with the current approach. I've seen it around and all the community is struggling with that and with the deprecation of kubelet containerized so 👍 for me